lib/rspec/mocks/method_double.rb

```diff
@@ -83,8 +83,11 @@ def restore_original_method
         return unless @method_is_proxied
 
         remove_method_from_definition_target
-        @method_stasher.restore if @method_stasher.method_is_stashed?
-        restore_original_visibility
+
+        if @method_stasher.method_is_stashed?
+          @method_stasher.restore
+          restore_original_visibility
```

Comment:
Actually, we could remove all the checks in `restore_original_visibility` when this is only called when `@method_stasher.method_is_stashed?`